src/evaluate/evaluator/base.py

```diff
@@ -126,6 +126,36 @@ def _compute_time_perf(start_time: float, end_time: float, num_samples: int) ->
             "latency_in_seconds": latency_sample,
         }
 
+    @staticmethod
+    def _infer_device():
+        """Helper function to check if GPU or CPU is available for inference."""
+        # try infer with torch first
+        try:
+            import torch
+
+            if torch.cuda.is_available():
+                device = 0  # first GPU
+            else:
+                device = -1  # CPU
+        except ImportError:
+            # if not available try TF
+            try:
+                import tensorflow as tf
+
+                if len(tf.config.list_physical_devices("GPU")) > 0:
+                    device = 0  # first GPU
+                else:
+                    device = -1  # CPU
+            except ImportError:
+                device = -1
+
+        if device == -1:
```

Comment:
In the rare case that neither `torch` or `tensorflow` are installed, this will error out because `device` is not assigned a value. Would it make sense to have a warning in that case?

Reply:
See comment in test

Reply:
Ah sorry I missed that - all good then!